src/pages/main/HomePage/components/LnbList.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import styled, { css } from 'styled-components';
+
+import { LnbInfo } from '../constants';
+
+const LnbList = () => {
+  const [isHoverIcon, setIsHoverIcon] = useState<boolean[]>([false, false, false, false, false, false, false, false]);
+  const [isClickIcon, setIsClickIcon] = useState<boolean[]>([false, false, false, false, false, false, false, false]);
+
+  const handleIconClick = (id: number, identifier: string) => {
+    if (identifier === 'click') {
+      const temp = isClickIcon.map((_, index) => (index === id ? true : false));
+      setIsClickIcon(temp);
+    } else if (identifier === 'mouseOver') {
+      const temp = isHoverIcon.map((_, index) => (index === id || isClickIcon[index] ? true : false));
+      setIsHoverIcon(temp);
+    } else {
+      const temp = isHoverIcon.map(() => false);
+      setIsHoverIcon(temp);
+    }
+  };
```

Comment:
여기가 그 문제의 호버,클릭 기타 등등 컴포넌트였구나 ....
분기처리가 엄청나네요 .. 고생했다 ,,

여기 부분을 좀 더 가독성 좋게 하려면 switch-case문을 사용해봐도 좋아요 !
```suggestion
 const handleIconClick = (id: number, identifier: string) => {
  switch (identifier) {
    case 'click':
      const clickTemp = isClickIcon.map((_, index) => (index === id ? true : false));
      setIsClickIcon(clickTemp);
      break;
    case 'mouseOver':
      const hoverTemp = isHoverIcon.map((_, index) => (index === id || isClickIcon[index] ? true : false));
      setIsHoverIcon(hoverTemp);
      break;
    default:
      const defaultTemp = isHoverIcon.map(() => false);
      setIsHoverIcon(defaultTemp);
  }
};
```
이런식으로!

Reply:
확실히 switch문이 더 가독성이 좋아지네요!!!